dynamodb/src/main/scala/zio/dynamodb/DynamoDBQuery.scala

```diff
@@ -653,9 +661,14 @@ object DynamoDBQuery {
     filterExpression: Option[FilterExpression] = None,
     projections: List[ProjectionExpression] = List.empty, // if empty all attributes will be returned
     capacity: ReturnConsumedCapacity = ReturnConsumedCapacity.None,
-    select: Option[Select] = None                         // if ProjectExpression supplied then only valid value is SpecificAttributes
+    select: Option[Select] = None,                        // if ProjectExpression supplied then only valid value is SpecificAttributes
+    totalSegments: Int = 1                                // TODO: Greater than 0
```

Comment:
```suggestion
    totalSegments: Int = 1
```
Do we still need the TODO?

Reply:
I think it would be nice to use zio-prelude to ensure that someone can't call `.parallel` with a number less than 1. I just put the TODO here while I was thinking about it. We currently have an if that would catch negative numbers and would just do a normal scan but I think having something a little more type safe would be nice? What do you think?

Reply:
so my 50 cents - I think generally we have been careful not to impose too many dependencies on the users of library and to keep things as lightweight as possible. It would be interesting to hear what John thinks.

Reply:
I'm going to get rid of the TODO and we can look at this at a later point.